GitUI/BuildServerIntegration/BuildServerWatcher.cs

```diff
@@ -200,6 +201,30 @@ public IBuildServerCredentials GetBuildServerCredentials(IBuildServerAdapter bui
             }
         }
 
+        public string ReplaceVariables(string projectNames)
+        {
+            //Extract "name of repo" from remote url
+            string remoteName = Module.GetCurrentRemote();
+            if (remoteName.IsNullOrWhiteSpace())
+            {
+                //No remote for the branch, for instance submodule. Use first remote 
+                var remotes = Module.GetRemotes();
+                if (remotes.Length > 0)
+                {
+                    remoteName = remotes[0];
+                }
+            }
+            var remoteUrl = Module.GetSetting(string.Format(SettingKeyString.RemoteUrl, remoteName));
+            var repoName = Path.GetFileNameWithoutExtension(remoteUrl);
+
+            if (repoName.IsNullOrWhiteSpace())
+                return projectNames;
+
+            return projectNames
+                .Replace("%REPO_SHORTNAME_U%", repoName.ToUpper())
+                .Replace("%REPO_SHORTNAME%", repoName);
```

Comment:
Forgot to ask the first time, why do we need two variables - the upper cased one and not upper cased?

Reply:
It depends if the webserver for the buildserver is case sensitive or not.
At least for Bitbucket server, the url is often converted to lower case when cloing. So if the "repo_shortname" is an acronym, the name will be incorrect.

But it is not unreasonable to request that the url and the build projects are aligned.
Jenkins is case insensitive

So there is a reason, but probably not important.
I have not made up my mind how I want to present this feature, the valid argument against the _U variable is how to describe it.

Reply:
A little correction: Bitbucket is not a build server, TeamCity is.

I just checked out TC instance (v10.0.5) and it didn't mind lower cased project Id (from normal title case).

If you insist on having `%REPO_SHORTNAME_U%` then perhaps it should be `%REPO_SHORTNAME_UPPER%` (and format it with `ToUpperInvariant()`). 
I'm not totally convinced it is required but I haven't used anything except TC.

Reply:
Removing %REPO_SHORTNAME_U%
If the build server is not case sensitive, it does not matter.